### DISCUSSION TITLE
How to use flux to autoupdate flux

### DISCUSSION THREAD

**Discussion by rverma-dev** [fluxcd/flux2#1424] [General]
The autoupdate flux could be as simple as updating just the docker images which can be implemented using image reconciler but it can get complicated if there are other changes.

It would be nice if we can have some way of autoupdate flux.

**Comment by kingdonb** [comment]
See the "Automated Upgrades" subheading here: https://fluxcd.io/docs/installation/#bootstrap-upgrade

> You can automate the components manifest update with GitHub Actions and open a PR when there is a new Flux version available. For more details please see [Flux GitHub Action docs](https://github.com/fluxcd/flux2/tree/main/action).

Flux (for GitHub users at least) has documentation that will show you how to open a pull-request automatically, with the next release of Flux, whenever Flux is upgraded. When you merge that pull request, Flux upgrades itself.

Flux is indeed a little more complicated than garden-variety Kubernetes app which can be upgraded by tracking the images with semver release versions (as Flux can do for your applications via ImagePolicy, just as how you suggest). The main complication making it more difficult is since it is built on CRDs, which can require manual intervention in order to upgrade.

A recent release of Flux [upgraded from v1alpha1 to v1alpha2](https://github.com/fluxcd/flux2/discussions/1333) version of the Image APIs, as an example of a change which is more complicated. But this utility GitHub Action is very straightforward, please check it out!
